### ISSUE TITLE
Error in mangrove.js when providing both a provider and a signer for connect 

### ISSUE THREAD

**Issue by dontrolle** [mangrovedao/mangrove#20]
When you provide both a `signer` and constructed `provider` for `mangrove.js´ in 

```
const provider = new ethers.providers.JsonRpcProvider(process.env["..."]);
const mgv = await Mangrove.connect({ signer: nonceManager, provider: provider });
```

you get an error from inside mangrove.js because of a malformed request. Maybe it should not be possible to send a `provider` when `signer` also has a `provider`?

**Comment by adhusson** [comment]
Providing both seems impossible -- I think users should give a probider url + (privateKey/mnemonic/etc). I made it explicitly forbidden in 4aa52500ecbc754a248428baa26e37e8d28d8894. What do you think?

**Comment by espendk** [comment]
I found this interesting statement from ricmoo, the author of ethers.js:

> In ethers, there are 3 possible cases:
> 
>     Provider - read-only access
>     Signer (without a provider) - write-only access
>     Signer (with a provider) - read/write access
https://github.com/ethers-io/ethers.js/issues/252#issuecomment-410913959

So maybe providing a Signer (with a Provider attached) should be the recommended way of connecting to Mangrove with read/write privileges?

**Comment by espendk** [comment]
To be more explicit, I suggest we change the error when user provides both Signer and Provider to: "Cannot give both a provider and a signer. Try giving a signer equipped with a provider."

https://github.com/mangrovedao/mangrove/commit/4aa52500ecbc754a248428baa26e37e8d28d8894#diff-7dd004513cbae88e9dd96ac62c8681a4179e3e7a4a18bbf57f28b0fead192d5bR101

**Comment by espendk** [comment]
Fixed in #180 .
